name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/BitfinexQuoteFeed.java

```diff
@@ -0,0 +1,201 @@
+        {
+            errors.add(new IOException(MessageFormat.format(Messages.MsgMissingTickerSymbol, security.getName())));
+            return false;
+        }
+
+        List<LatestSecurityPrice> prices = getHistoricalQuotes(security, LocalDate.now(), errors);
+
+        if (prices.isEmpty())
+        {
+            return false;
+        }
+        else
+        {
+            LatestSecurityPrice price = prices.get(prices.size() - 1);
+            if (price.getValue() != 0)
+                security.setLatest(price);
+
+            return true;
+        }
+    }
+
+    @Override
+    public boolean updateHistoricalQuotes(Security security, List<Exception> errors)
+    {
+        if (security.getTickerSymbol() == null)
+        {
+            errors.add(new IOException(MessageFormat.format(Messages.MsgMissingTickerSymbol, security.getName())));
+            return false;
+        }
+
+        LocalDate quoteStartDate = LocalDate.MIN;
+
+        if (!security.getPrices().isEmpty())
+            quoteStartDate = security.getPrices().get(security.getPrices().size() - 1).getDate();
+
+        List<LatestSecurityPrice> prices = getHistoricalQuotes(security, quoteStartDate, errors);
+
+        boolean isUpdated = false;
+        for (SecurityPrice p : prices)
+        {
+            if (p.getDate().isBefore(LocalDate.now()))
+            {
+                boolean isAdded = security.addPrice(p);
+                isUpdated = isUpdated || isAdded;
+            }
+        }
+        return isUpdated;
+    }
+    
+    @SuppressWarnings("unchecked")
+    private List<LatestSecurityPrice> convertBitfinexJsonArray(JSONArray ohlcArray, List<Exception> errors)
+    {
+        final Long secondsPerDay = 24L * 60 * 60;
+        List<LatestSecurityPrice> prices = new ArrayList<>();
+        
+        if(ohlcArray.size() == 0)
+            return prices;
+        
+        if (ohlcArray.get(0) instanceof JSONArray)
+        {
+            // ohlcArray is an array of quotes
+            ohlcArray.forEach(e -> {
+                JSONArray quoteEntry = (JSONArray) e;
+                Long timestamp = Long.parseLong(quoteEntry.get(0).toString());
+
+                try
+                {
+                    Long open = YahooHelper.asPrice(quoteEntry.get(1).toString());
+                    Long close = YahooHelper.asPrice(quoteEntry.get(2).toString());
+                    Long high = YahooHelper.asPrice(quoteEntry.get(3).toString());
+                    Long low = YahooHelper.asPrice(quoteEntry.get(4).toString());
+                    Integer volume = YahooHelper.asNumber(quoteEntry.get(5).toString());
+
+                    LatestSecurityPrice price = new LatestSecurityPrice(LocalDate.ofEpochDay(timestamp / 1000 / secondsPerDay), close);
+                    price.setHigh(high);
+                    price.setLow(low);
+                    price.setVolume(volume);
+                    price.setPreviousClose(open);
+
```

Comment:
There is no need to store all high, low, volume information as is blow up the data storage. Only the latest is needed. ```SecurityPrice``` should be preferable.

Reply:
When there is no need for full OHLCV data, why is there an extended class for this?
If we only should use SecurityPrice class, the open, high, low price should also be removed from KrakenQuoteFeed and AlphaVantageQuoteFeed

Reply:
The currently implemented logic is:
If it is the "latest price" then it contains high, low, etc.
For historical quotes, PP only stores the "daily closing price".

The rationale is: to calculate past performance, only the closing price is used. To store high, lows,volumne in XML, it would only blow up the XML. PP does not intent to be a charting tool - there are many (and better) out there.

I will fix ```KrakenQuoteFeed``` as well b/c it does it wrong.

```AlphavantageQuoteFeed``` passes the type of price class to be created to the method:
```java
T price = klass.getConstructor().newInstance();
```

Reply:
I will keep this in mind! Thank you for the explanation.

Reply:
FIY - I am in the process of merging the PR.